gRPC/Client Examples/RealtimeGraphing/BICManager.cs

```diff
@@ -143,18 +154,61 @@ public bool BICConnect()
             Console.WriteLine("Connecting to implantable device.");
             var connectDeviceReply = deviceClient.ConnectDevice(new ConnectDeviceRequest() { DeviceAddress = DeviceName, LogFileName = "./deviceLog.txt" });
 
+            // Get electrode impedances
+            for (uint channelNum = 0; channelNum < numSensingChannelsDef; channelNum++)
+            {
+                bicGetImpedanceReply chanImpedValue = deviceClient.bicGetImpedance(new bicGetImpedanceRequest() { DeviceAddress = DeviceName, Channel = channelNum });
+                //Console.WriteLine("CH " + (channelNum + 1).ToString() + ": " + chanImpedValue.Success);
+                if (chanImpedValue.Success == "success")
+                {
+                    // add impedance value and units to buffer
+                    impedBuffer.Add(Math.Ceiling(chanImpedValue.ChannelImpedance).ToString() + " " + chanImpedValue.Units);
+                }
+                else
+                {
+                    // note error in retrieiving impedance 
+                    impedBuffer.Add(chanImpedValue.Success);
+                }
+            }
+
+            // Check that impedances can be logged
+            if (File.Exists(impedFilePath))
+            {
+                File.Delete(impedFilePath);
+            }
+            impedFileStream = new FileStream(impedFilePath, FileMode.Create, FileAccess.Write, FileShare.None, 4096, FileOptions.Asynchronous);
+            impedFileWriter = new StreamWriter(impedFileStream);
+            string impedEntry = "";
+            for (int chNum = 0; chNum < impedBuffer.Count; chNum++)
+            {
+                impedEntry = "CH" + (chNum + 1).ToString();
+                impedEntry += ", " + impedBuffer[chNum];
+                impedFileWriter.WriteLine(impedEntry);
+            }
+
+            // Close impedance logging items
+            impedFileWriter.Flush();
+            impedFileWriter.Dispose();
+            impedFileStream.Dispose();
+
             // Start up the neural stream
             neuroMonitor = Task.Run(neuralMonitorTaskAsync);
 
+            // Start up the connection stream
```

Comment:
did the EP app not have the reconnecting logic added?

Reply:
this is why in theory it may be advantageous to have some of the reconnecting logic in the microservice